modal/dict.py

```diff
@@ -143,7 +143,11 @@ async def __getitem__(self, key: Any) -> Any:
 
         This function only works in a synchronous context.
         """
-        return await self.get(key)
+        value = await self.get(key)
```

Comment:
this is a bit of an edge case but the user might store `None` in the dictionary and we shouldn't raise a `KeyError` on that

```
>>> d = {"foo": None}
>>> print(d["foo"])
None
```

You might want to define a custom sentinel e.g. `NOT_FOUND = object()` in the global scope and do this:

```python
value = await self.get(key, NOT_FOUND)
if value is NOT_FOUND:
    raise KeyError(...)
```